core/src/main/java/com/amplifyframework/auth/AuthUserAttributeKey.java

```diff
@@ -0,0 +1,225 @@
+        return new AuthUserAttributeKey(LOCALE);
+    }
+
+    /**
+     * Factory method for getting middle name attribute.
+     * @return a pre-configured middle name attribute
+     */
+    public static AuthUserAttributeKey middleName() {
+        return new AuthUserAttributeKey(MIDDLE_NAME);
+    }
+
+    /**
+     * Factory method for getting name attribute.
+     * @return a pre-configured name attribute
+     */
+    public static AuthUserAttributeKey name() {
+        return new AuthUserAttributeKey(NAME);
+    }
+
+    /**
+     * Factory method for getting nickname attribute.
+     * @return a pre-configured nickname attribute
+     */
+    public static AuthUserAttributeKey nickname() {
+        return new AuthUserAttributeKey(NICKNAME);
+    }
+
+    /**
+     * Factory method for getting phone number attribute.
+     * @return a pre-configured phone number attribute
+     */
+    public static AuthUserAttributeKey phoneNumber() {
+        return new AuthUserAttributeKey(PHONE_NUMBER);
+    }
+
+    /**
+     * Factory method for getting picture attribute.
+     * @return a pre-configured picture attribute
+     */
+    public static AuthUserAttributeKey picture() {
+        return new AuthUserAttributeKey(PICTURE);
+    }
+
+    /**
+     * Factory method for getting preferred username attribute.
+     * @return a pre-configured preferred username attribute
+     */
+    public static AuthUserAttributeKey preferredUsername() {
+        return new AuthUserAttributeKey(PREFERRED_USERNAME);
+    }
+
+    /**
+     * Factory method for getting profile attribute.
+     * @return a pre-configured profile attribute
+     */
+    public static AuthUserAttributeKey profile() {
+        return new AuthUserAttributeKey(PROFILE);
+    }
+
+    /**
+     * Factory method for getting updated at attribute.
+     * @return a pre-configured updated at attribute
+     */
+    public static AuthUserAttributeKey updatedAt() {
+        return new AuthUserAttributeKey(UPDATED_AT);
+    }
+
+    /**
+     * Factory method for getting website attribute.
+     * @return a pre-configured website attribute
+     */
+    public static AuthUserAttributeKey website() {
+        return new AuthUserAttributeKey(WEBSITE);
+    }
+
+    /**
+     * Factory method for getting zone info attribute.
+     * @return a pre-configured zone info attribute
+     */
+    public static AuthUserAttributeKey zoneInfo() {
+        return new AuthUserAttributeKey(ZONE_INFO);
+    }
+
+    /**
+     * Factory method for creating your own custom user attribute.
+     * @param attributeKey The key for the custom user attribute
+     * @return a custom provider
+     */
+    public static AuthUserAttributeKey custom(String attributeKey) {
+        return new AuthUserAttributeKey(attributeKey);
+    }
+
+    /**
+     * Returns the String key for the attribute.
+     * @return the String key for the attribute
+     */
+    @NonNull
+    public String getKeyString() {
```

Comment:
`getAttributeKey()` , to match the field? Or, change the field name to match the method name?

Reply:
This was another thing I debated quite a bit. The reason for choosing this name is the following - if the user was wanting to get the string value of the attribute key, it would look like: myAttributeVariable.getKey().getAttributeKey() which seems very duplicative and confusing. 

I thought in this case it would be clearer what that method does naming it this way so the user would be doing myAttributeVariable.getKey().getKeyString()